LeaderboardBackend.Test/Features/Users/LoginTests.cs

```diff
@@ -0,0 +1,126 @@
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Net.Http.Json;
+using System.Threading.Tasks;
+using LeaderboardBackend.Models.Entities;
+using LeaderboardBackend.Models.Requests;
+using LeaderboardBackend.Models.Validation;
+using LeaderboardBackend.Test.Fixtures;
+using LeaderboardBackend.Test.Lib;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.DependencyInjection;
+using NUnit.Framework;
+
+namespace LeaderboardBackend.Test.Features.Users;
+
+public class LoginTests : IntegrationTestsBase
+{
+    private const string LOGIN_URI = "/login";
+    private const string VALID_EMAIL = "valid@user.com";
+    private const string BANNED_EMAIL = "banned@user.com";
+    private const string VALID_PASSWORD = "P4ssword";
+
+    [OneTimeSetUp]
+    public void Init()
+    {
+        s_factory.ResetDatabase();
+
+        // TODO: Swap to creating users via the UserService instead of calling the DB, once
+        // it has the ability to change a user's roles.
+        using IServiceScope s = s_factory.Services.CreateScope();
+        ApplicationContext dbContext = s.ServiceProvider.GetRequiredService<ApplicationContext>();
+        dbContext.Users.AddRange(new[]
+        {
+            new User{
+                Email = VALID_EMAIL,
+                Password = BCrypt.Net.BCrypt.EnhancedHashPassword(VALID_PASSWORD),
+                Username = "Test_User",
+            },
+            new User{
+                Email = BANNED_EMAIL,
+                Password = BCrypt.Net.BCrypt.EnhancedHashPassword(VALID_PASSWORD),
+                Role = UserRole.Banned,
+                Username = "Banned_User",
+            },
+        });
+        dbContext.SaveChanges();
+    }
+
+    [Test]
+    public async Task Login_ValidRequest_ReturnsLoginResponse()
+    {
+        LoginRequest request = new()
+        {
+            Email = TestInitCommonFields.Admin.Email,
+            Password = TestInitCommonFields.Admin.Password,
+        };
+
+        HttpResponseMessage res = await Client.PostAsJsonAsync(LOGIN_URI, request);
+
+        res.Should().HaveStatusCode(HttpStatusCode.OK);
+        LoginResponse? content = await res.Content.ReadFromJsonAsync<LoginResponse>();
+        content.Should().NotBeNull();
+        content!.Token.Should().MatchRegex(@"^[A-Za-z0-9-_=]{36}\.[A-Za-z0-9-_=]{190}\.[A-Za-z0-9-_=]{43}$");
```

Comment:
Use this to check if it is a JWT `Jwt.SecurityTokenHandler.CanReadToken()`

Or even better, validate it: https://github.com/leaderboardsgg/leaderboard-backend/blob/48e31ccfb865b6f8e3f02c1fdb3d2a61b7de8156/LeaderboardBackend/Authorization/UserTypeAuthorizationHandler.cs#L102